internal/resources/cloudprovider/resource_aws_account.go

```diff
@@ -82,6 +82,9 @@ func (r *resourceAWSAccount) Schema(ctx context.Context, req resource.SchemaRequ
 			"resource_id": schema.StringAttribute{
 				Description: "The ID given by the Grafana Cloud Provider API to this AWS Account resource.",
 				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
```

Comment:
what's this again? didn't quite get it

Reply:
let me put a doc link as a comment on this one since it's kind of obscure: https://developer.hashicorp.com/terraform/plugin/framework/resources/plan-modification#usestateforunknown

The resource_id should never change for an account, it can't be updated, it's assigned by our API. However, the schema for this attribute must be marked `Computable` because its value can only come from the API, it can't be specified by the requestor.

 Because of `Computable`, with no value being assigned by the requestor, every single modification to an account resource in Terraform would also mark the `resource_id` as "(known after apply)" in the plan. But this is unnecessary, because we know that the `resource_id` can never actually change. 
 
This framework-standard modifier makes it so that whenever the `resource_id` is unknown (which is on creation, and every time an existing resource is to be updated), it will fall back to the previously known state value. Therefore, on creation, the state is nonexistent for the resource, so indeed it will show "(known after apply)" in the plan, but after creation, on modification, the resource_id is known, so it will just keep assuming that same resource_id value in the plan without changing.